### PR TITLE
Add support for mysql backend

### PR DESCRIPTION
**Pull request by tsteur**

### CODE REVIEW

File: Tasks.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * run eg using ./console core:run-scheduled-tasks "Piwik\Plugins\QueuedTracking\Tasks.optimizeQueueTable"
+     */
+    public function optimizeQueueTable()
+    {
+        $settings = Queue\Factory::getSettings();
+        if ($settings->isMysqlBackend() && $settings->queueEnabled->getValue()) {
+
+            $db = Db::get();
+            $prefix = Common::prefixTable(MySQL::QUEUED_TRACKING_TABLE_PREFIX);
+            $tables = $db->fetchCol("SHOW TABLES LIKE '" . $prefix . "%'");
```
Comment (tsteur) [inline Tasks.php]: fyi @mattab also added an optimize table to improve storage space.

File: Queue/Backend/MySQL.php [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+            return 0;
+        }
+
+        if (empty($row['expiry_time'])) {
+            // key exists but has no associated expire
+            return 99999999;
+        }
+
+        $secondsLeft = $row['expiry_time'] - $row['timestamp'];
+        if ($secondsLeft < 0) {
```
[the remark this answers is not in the document]
Comment (tsteur) [reply Queue/Backend/MySQL.php]: Can change it, would return `0` in either case.

File: Queue/Lock.php [hunk 3]
```diff
@@ -78,6 +78,11 @@ public function expireLock($ttlInSeconds)
 
             if (!$success) {
                 $value = $this->backend->get($this->lockKey);
+                if ($value === $this->lockValue) {
+                    // the expire did not work because the same time was already set and we just tried to set the same ttl
+                    // again too fast within one second
+                    return true;
+                }
```
[the remark this answers is not in the document]
Comment (tsteur) [reply Queue/Lock.php]: It does't really affect it, so will change it 👍  The reason this doesn't affect redis btw is that we use seconds for the TTL, but redis basically `milliseconds`.

File: Queue/Backend/MySQL.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (diosmosis) [reply Queue/Backend/MySQL.php]: Just wondering if it would be safer to do `<= 0`, but not as knowledgeable of the specifics w/ regard to QueuedTracking

**Comment by diosmosis** [comment]
FYI, UI tests failing

**Comment by tsteur** [comment]
just pushing the fixed ui tests

**Comment by diosmosis** [comment]
argh, now there's a conflict. I can fix them myself tomorrow morning & merge this

**Pull request merged**